File: java/sadporm/src/main/java/com/sadp/orm/repo/UserRepository.java
/**
 * 
 */
package com.sadp.orm.repo;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.sadp.orm.dto.User;


/**
 * @author ntanwa
 *
 */
@Repository
public interface UserRepository  extends JpaRepository<User, Long> {
	List<User> findAll();
}

File: java/userservice/src/main/kotlin/com/nstane/userservice/repositories/UserRepository.kt
package com.nstane.userservice.repositories

import com.nstane.userservice.entities.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository: JpaRepository<User, Long> {

    @Query("Select u from User u where u.userName = ?1 ")
    fun finByUserName(userName: String?): Optional<User>
}
